### PR TITLE
Stream window test fail fix

### PR DESCRIPTION
**Pull request by johanandren**
Make sure the stream window resize always happens.
Previously it would only be triggered in Akka 2.6 by a earlier pull causing the test to fail
while with Akka 2.5 would pass. This adds an explicit pull to trigger the same behavior on both.

Credits for fix mostly goes to jrudolph who helped me a lot.

References #3606

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Seems OK, though it's a bit unfortunate we're now adding overly-specific 'noise' to this test: we don't care how many window updates we get, we only care that we get a RST when we exceed the window. You'd almost want something like `fishForMessage` in Akka, though that is not super nice to use either...

In any case this :+1: !

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 1]
```diff
@@ -409,8 +409,10 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
         // trigger a connection-level WINDOW_UPDATE
         sendDATA(TheStreamId, endStream = false, ByteString("0000"))
         expectWindowUpdate()
+        entityDataIn.expectUtf8EncodedString("0000")
+        expectWindowUpdate() // window resize/update triggered
 
-        sendFrame(DataFrame(TheStreamId, endStream = false, ByteString("0" * 100000)))
+        sendFrame(DataFrame(TheStreamId, endStream = false, ByteString("0" * 512001)))
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:415]:
```suggestion
        sendFrame(DataFrame(TheStreamId, endStream = false, ByteString("0" * 512001))) // more than default `incoming-stream-level-buffer-size = 512kB`
```

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 2]
```diff
@@ -409,8 +409,10 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
         // trigger a connection-level WINDOW_UPDATE
         sendDATA(TheStreamId, endStream = false, ByteString("0000"))
         expectWindowUpdate()
+        entityDataIn.expectUtf8EncodedString("0000")
+        expectWindowUpdate() // window resize/update triggered
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:413]:
```suggestion
        entityDataIn.expectUtf8EncodedString("0000")
        pollForWindowUpdates(500.millis) // window resize/update triggered
```

**Review by jrudolph — APPROVED** [review]
LGTM with a suggestion following @raboof's comment.

**Comment by jrudolph** [comment]
> Seems OK, though it's a bit unfortunate we're now adding overly-specific 'noise' to this test: we don't care how many window updates we get, we only care that we get a RST when we exceed the window. 

Partly it's because of https://github.com/akka/akka-http/issues/3610.

> You'd almost want something like `fishForMessage` in Akka, though that is not super nice to use either...

Yes, good point, we have that :) See above.

**Pull request merged**